test/pool.test.ts

```diff
@@ -72,6 +86,51 @@ abstract class PoolUser {
     this.addr = addr;
   }
 
+  abstract getBalance(): Promise<BigNumber>;
+
+  abstract submitTopUp(amount: number): Promise<ContractTransaction>;
+
+  async collect(expectedAmount: number): Promise<void> {
+    await this.expectCollectableOnNextBlock(expectedAmount);
+    const balanceBefore = await this.getBalance();
+    await submit(this.pool.collect({ gasPrice: 0 }), "collect");
+    const balanceAfter = await this.getBalance();
+    const collected = balanceAfter.sub(balanceBefore).toNumber();
+    expect(collected).to.equal(
+      expectedAmount,
+      "The collected amount is different from the expected amount"
+    );
```

Comment:
Making assertions on a balance change seems to be a common pattern. How about this.
```suggestion
    await this.expectBalanceChange(
      () => {
        await submit(this.pool.collect({ gasPrice: 0 }), "collect");
      },
      expectedAmount,
      "The collected amount is different from the expected amount"
    );
```

Reply:
Great catch, applied